### PR TITLE
Vendordep Templating.

### PR DESCRIPTION
**Pull request by thenetworkgrinch**
Outputs filled out vendordep file to `$buildDir/repos/` with filled in variables from the gradle config.

### CODE REVIEW

File: publish.gradle [hunk 1]
```diff
@@ -156,6 +161,16 @@ publishing {
...
+// Apply template variables from the vendordep file.
+// Replaces ${VARIABLE} with VARIABLE: value in expand()
+copy {
+    from templateVendorFile
+    into "$releasesRepoUrl/../"
+    expand( version: pubVersion,
+            groupId: artifactGroupId,
+            artifactId: baseArtifactId)
+}
+
```
Comment (rzblue) [inline publish.gradle]: This occurs at configuration time, not build time

**Review by rzblue — CHANGES_REQUESTED** [review]
It would be better to generate this as an artifact that gets published

**Comment by thenetworkgrinch** [comment]
Like this? I am not sure how to add it to `MavenPublication`. I tried doing
```
vendordep(MavenPublication) {
                artifact vendordepJson

                artifactId = "${baseArtifactId}-vendordep"
                groupId artifactGroupId
                version pubVersion
            }
```
but it didnt recognize the artifact type as valid even though it outputs a file.

**Comment by thenetworkgrinch** [comment]
I just made a zip task which wraps up the generated vendordep and is able to publish it to maven. 

I still dont think this is necessary or a good idea? Why would you want to publish your vendordep to maven when its supposed to be hosted and accessible?

The current version where it outputs the vendordep to `$buildDir` seems fine and really would be better off if it outputted to `$buildDir/repos`. That way vendors can copy `$buildDir/repos` over to their webhost and have it be done.

File: publish.gradle [hunk 2]
```diff
@@ -156,6 +188,7 @@ publishing {
     }
 }
 
+
```
Comment (ThadHouse) [inline publish.gradle]: Nit. Extra space

File: publish.gradle [hunk 3]
```diff
@@ -200,6 +233,14 @@ model {
                 groupId artifactGroupId
                 version pubVersion
             }
+
+            vendordep(MavenPublication) {
+                artifact vendordepJsonZip
+
+                artifactId = "${baseArtifactId}-vendordep"
```
Comment (ThadHouse) [inline publish.gradle:240]: You can actually publish the vendordep json directly. No need to zip it.

File: publish.gradle [hunk 4]
```diff
@@ -127,6 +132,33 @@ task outputJavadocJar(type: Jar, dependsOn: javadoc) {
...
+        expand(version: pubVersion,
+                groupId: artifactGroupId,
+                artifactId: baseArtifactId)
+    }
+}
+
+task vendordepJsonZip(type: Zip) {
+    destinationDirectory = outputsFolder
+    archiveBaseName = templateVendorFile
+    archiveClassifier = "sources"
```
Comment (ThadHouse) [inline publish.gradle]: This classifier is wrong, but with my other comment this zip task can go away completely.

File: publish.gradle (hunk 3 above)
Comment (thenetworkgrinch) [reply publish.gradle:240]: How would i? I had tried and it would constantly fail.

Comment (thenetworkgrinch) [reply publish.gradle:240]: Do i even really need to. I only added this in response to @rzblue I dont really think its necessary

Comment (ThadHouse) [reply publish.gradle:240]: I guess the zip is fine. So I'll add some other comments on things to fix.

File: publish.gradle (hunk 4 above)
Comment (ThadHouse) [reply publish.gradle]: See other comments.

File: publish.gradle [hunk 5]
```diff
@@ -127,6 +132,33 @@ task outputJavadocJar(type: Jar, dependsOn: javadoc) {
...
+        into "$buildDir/repos/"
+        expand(version: pubVersion,
+                groupId: artifactGroupId,
+                artifactId: baseArtifactId)
+    }
+}
+
+task vendordepJsonZip(type: Zip) {
+    destinationDirectory = outputsFolder
+    archiveBaseName = templateVendorFile
```
Comment (ThadHouse) [inline publish.gradle]: Remove archiveClassifier, replace archiveBaseName with `archiveBaseName = "vendordepjson"`

**Comment by ThadHouse** [comment]
On lines 172 and 177 of build.gradle, add `addTaskToCopyAllOutputs(vendordepJsonZip)` and `build.dependsOn vendordepJsonZip`

**Review by ThadHouse — APPROVED** [review]

**Review by ThadHouse — APPROVED** [review]

**Pull request merged**